test/torchaudio_unittest/functional/functional_cpu_test.py

```diff
@@ -70,46 +70,124 @@ def test_pitch(self, frequency):
-        x2 = F.DB_to_amplitude(db, ref, power)
-
-        torch.testing.assert_allclose(x2, torch.abs(x), atol=5e-5, rtol=1e-5)
+class Testamplitude_to_DB(common_utils.TorchaudioTestCase):
+    def _ensure_reversible(self, spec):
+        """Check `amplitude_to_db` returns the original when reversed."""
+        AMPLITUDE_MULT = 20.
+        POWER_MULT = 10.
+        AMIN = 1e-10
+        REF = 1.0
+        DB_MULT = math.log10(max(AMIN, REF))
 
         # Spectrogram amplitude -> DB -> amplitude
-        db = F.amplitude_to_DB(spec, multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
+        db = F.amplitude_to_DB(spec, AMPLITUDE_MULT, AMIN, DB_MULT, top_db=None)
+        x2 = F.DB_to_amplitude(db, REF, 0.5)
 
         torch.testing.assert_allclose(x2, spec, atol=5e-5, rtol=1e-5)
 
-        # Waveform power -> DB -> power
-        multiplier = 10.
-        power = 1.
-
-        db = F.amplitude_to_DB(x, multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
-
-        torch.testing.assert_allclose(x2, torch.abs(x), atol=5e-5, rtol=1e-5)
-
         # Spectrogram power -> DB -> power
-        db = F.amplitude_to_DB(spec, multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
-
-        torch.testing.assert_allclose(x2, spec, atol=5e-5, rtol=1e-5)
+        db = F.amplitude_to_DB(spec, POWER_MULT, AMIN, DB_MULT, top_db=None)
+        x2 = F.DB_to_amplitude(db, REF, 1.)
+
+        self.assertEqual(x2, spec)
+
+    def test_amplitude_to_DB_batch(self):
+        spec = torch.rand([2, 2, 100, 100]) * 200
+        self._ensure_reversible(spec)
+
+    def test_amplitude_to_DB_channels(self):
+        spec = torch.rand([2, 100, 100]) * 200
+        self._ensure_reversible(spec)
+
+    def test_amplitude_to_DB_freq(self):
+        spec = torch.rand([100, 100]) * 200
+        self._ensure_reversible(spec)
+
+    def _check_top_db(self, spec):
+        AMPLITUDE_MULT = 20.
+        AMIN = 1e-10
+        REF = 1.0
+        DB_MULT = math.log10(max(AMIN, REF))
+        TOP_DB = 40.
+
+        decibels = F.amplitude_to_DB(spec, AMPLITUDE_MULT, AMIN,
+                                     DB_MULT, top_db=TOP_DB)
+        above_top = decibels >= 6.0205
+        assert above_top.all(), decibels
+
+    def test_top_db_batch(self):
+        spec = torch.rand([1, 2, 100, 100]) * 200
+        # Predictability
+        spec[0, 0, 1] = 0
+        spec[0, 0, 0] = 200
+        self._check_top_db(spec)
+
+    def test_top_db_channel(self):
+        spec = torch.rand([1, 2, 100, 100]) * 200
+        # Predictability
+        spec[0, 0, 1] = 0
+        spec[0, 0, 0] = 200
+        self._check_top_db(spec[0])
+
+    def test_top_db_freq(self):
+        spec = torch.rand([1, 2, 100, 100]) * 200
+        # Predictability
+        spec[0, 0, 1] = 0
+        spec[0, 0, 0] = 200
+        self._check_top_db(spec[0][0])
+
+    def test_batched(self):
```

Comment:
I think `test_batched` and `test_per_spectrogram` are better suited in `batch_consistency_test`. And we should be applying the second type of test to all the functions.

Reply:
I've moved these verbatim to `batch_consistency_test.py` - is that OK?